### PR TITLE
Server

### PR DESCRIPTION
**Pull request by chriztso**
Changes
1. Edited server code
2. Edited client's get request and set state function

### CODE REVIEW

File: server2.js [hunk 1]
```diff
@@ -1,75 +1,42 @@
...
-})
+app.listen(PORT);
 
-app.get('/api/menu/:meal/:id', (req, res) =>{
-    var type = req.params.meal;
-    var id = req.params.id;   
-    let queryStr = `select * from ${type} WHERE id = ${id}`;
-    connection.connection.query(queryStr, function (err, result) {
+app.get('/api/menu/:meal/:id', (req, res) => {
+    connection.query(`select * from ${req.params.meal} WHERE restaurant_id = ${req.params.id}`, function (err, result) {
```
Comment (ecuyle) [inline server2.js]: I would recommend using Postgres' template query strings to sanitize inputs. (ie `SELECT * FROM table WHERE restaurant_id=$1`)

File: server2.js [hunk 2]
```diff
@@ -1,75 +1,42 @@
...
- let queryStr = "select * from " + type;
- connection.connection.query(queryStr, function (err, result) {
-   if (err) {
-     console.log("error message: ", err);
-     return;
-   }
-   res.send(result);
- });
-})
+app.listen(PORT);
```
Comment (ecuyle) [inline server2.js]: Best practice is normally to put`app.listen` at the bottom of a server file after all the routes are set up. Just a suggestion!

File: server2.js [hunk 3]
```diff
@@ -1,75 +1,42 @@
...
-    connection.connection.query(query, function (err, result) {
-      if (err) {
-        console.log("error message: ", err);
-        return;
-      }
-      res.send(result);
-    });
+app.post('/api/menu/:meal', (req, res) => {
+  var type = req.params.meal;
+  var query = `INSERT INTO ${type} (restaurant_id, name, description, price) VALUES (${req.body.restaurant_id}, '${req.body.name}', '${req.body.description}', ${req.body.price})`;
```
Comment (ecuyle) [inline server2.js]: Same comment as above for template query strings

**Review by ecuyle — CHANGES_REQUESTED** [review]
Minor changes requested regarding template query strings when making DB calls. Good work so far!

File: server2.js (hunk 1 above)
Comment (chriztso) [reply server2.js]: Thanks for the suggestions Eric. I have made those changes, let me know what you think.

**Review by ecuyle — APPROVED** [review]
LGTM!

**Pull request merged**